### PR TITLE
Custom Color support

### PR DESCRIPTION
**Pull request by S4mpsa**

### CODE REVIEW

File: hud/init.lua [hunk 1]
```diff
@@ -32,8 +32,8 @@ local function save()
     powerDisplayUsers = {}
     toolbarUsers = {}
     for address, data in pairs(glassData) do
-        if data.energyDisplay then table.insert(powerDisplayUsers, {component.proxy(address), {data.xRes, data.yRes}, data.scale, data.borderColor, data.primaryColor, data.accentColor}) end
-        if data.toolbar then table.insert(toolbarUsers, {component.proxy(address), {data.xRes, data.yRes}, data.scale, data.offset, data.borderColor, data.primaryColor, data.accentColor}) end
+        if data.energyDisplay then table.insert(powerDisplayUsers, {component.proxy(address), {data.xRes or 2560, data.yRes or 1440}, data.scale or 3, data.backgroundColor or colors.darkGray, data.primaryColor or colors.electricBlue, data.accentColor or colors.accentColor}) end
+        if data.toolbar then table.insert(toolbarUsers, {component.proxy(address), {data.xRes, data.yRes}, data.scale, data.offset, data.backgroundColor, data.primaryColor, data.accentColor}) end
```
Comment (gordominossi) [inline hud/init.lua:36]: Didn't they have a default value somewhere else?

File: hud/toolbar.lua [hunk 2]
```diff
@@ -78,15 +78,14 @@ function toolbar.widget(glasses)
...
         hudObjects[i].dynamic.clock.setText(timeString)
-        --Following is temporary
-        local maxMemory = computer.totalMemory()
-        local usedMemory = maxMemory - computer.freeMemory()
-        hudObjects[i].dynamic.memory.setText("Memory used: "..parser.percentage(usedMemory/maxMemory))
-        --
+        --Uncomment to debug memory usage
+        --local maxMemory = computer.totalMemory()
+        --local usedMemory = maxMemory - computer.freeMemory()
+        --hudObjects[i].dynamic.memory.setText("Memory used: "..parser.percentage(usedMemory/maxMemory))
```
Comment (gordominossi) [inline hud/toolbar.lua:88]: Could be an `if`

File: hud/init.lua (hunk 1 above)
Comment (S4mpsa) [reply hud/init.lua:36]: They do, this is spaghetti code that fixed a crash I was having that I never properly fixed.

File: server/usecases/get-lsc-status.lua [hunk 3]
```diff
@@ -28,13 +28,12 @@ local function exec(address, name)
...
+
     local status = {
-        storedEU = math.floor(string.gsub(lsc.getSensorInformation()[2], "([^0-9]+)", "")),
-        EUCapacity = math.floor(string.gsub(lsc.getSensorInformation()[3], "([^0-9]+)", "") + 0),
+        storedEU = math.floor(string.gsub(sensorInformation[2], "([^0-9]+)", "")),
+        EUCapacity = math.floor(string.gsub(sensorInformation[3], "([^0-9]+)", "") + 0),
         problems = problems,
-        passiveLoss = lsc:getWorkMaxProgress() and
-            parser.getInteger(sensorInformation[4]) or 0,
+        passiveLoss = parser.getInteger(sensorInformation[4]) or 0,
```
Comment (gordominossi) [inline server/usecases/get-lsc-status.lua:36]: :+1:

File: lib/graphics/gui.lua [hunk 4]
```diff
@@ -299,28 +300,40 @@ function gui.colorSelection(x, y, colorList)
...
+    local height = #colorTable + 4
     if maxX <= (x+longestName+5) then x = maxX-(longestName+4) end
     if maxY <= (y+height) then y = maxY-(height-1) end
     local page = gpu.allocateBuffer(longestName+5, height)
     gpu.setActiveBuffer(page)
-    graphics.rectangle(1, 1, 5+longestName, 4+2*#colorTable, borderColor)
-    graphics.rectangle(5, 3, longestName, 2*#colorTable, colors.black)
+    graphics.rectangle(1, 1, 5+longestName, 6+2*#colorTable, borderColor)
+    graphics.rectangle(5, 3, longestName, 2+2*#colorTable, colors.black)
+    graphics.text(5, 3, "Custom")
```
Comment (gordominossi) [inline lib/graphics/gui.lua:310]: NICE

**Pull request merged**